servicetalk-test-resources/src/test/java/io/servicetalk/test/resources/DefaultTestCertsTest.java

```diff
@@ -17,76 +17,109 @@
-        assertEquals(1366, contents.length());
+        loadCertificate(DefaultTestCerts.loadServerCAPem());
     }
 
     @Test
     void loadServerP12() throws Exception {
-        String contents = readFully(DefaultTestCerts.loadServerP12());
-        assertEquals(4262, contents.length());
+        loadKeystore(DefaultTestCerts.loadServerP12());
     }
 
     @Test
     void loadClientKey() throws Exception {
-        String contents = readFully(DefaultTestCerts.loadClientKey());
-        assertEquals(2483, contents.length());
+        loadPrivateKey(DefaultTestCerts.loadClientKey());
     }
 
     @Test
     void loadClientPem() throws Exception {
-        String contents = readFully(DefaultTestCerts.loadClientPem());
-        assertEquals(1337, contents.length());
+        loadCertificate(DefaultTestCerts.loadClientPem());
     }
 
     @Test
     void loadClientCAPem() throws Exception {
-        String contents = readFully(DefaultTestCerts.loadClientCAPem());
-        assertEquals(1366, contents.length());
+        loadCertificate(DefaultTestCerts.loadClientCAPem());
     }
 
     @Test
     void loadClientP12() throws Exception {
-        String contents = readFully(DefaultTestCerts.loadClientP12());
-        assertEquals(4262, contents.length());
+        loadKeystore(DefaultTestCerts.loadClientP12());
     }
 
     @Test
     void loadTrustP12() throws Exception {
-        String contents = readFully(DefaultTestCerts.loadTruststoreP12());
-        assertEquals(2422, contents.length());
+        loadKeystore(DefaultTestCerts.loadTruststoreP12());
     }
 
-    private String readFully(final InputStream inputStream) throws IOException {
-        try (BufferedReader buffer = new BufferedReader(new InputStreamReader(inputStream, US_ASCII))) {
-            return buffer.lines().collect(joining("\n"));
+    private Certificate loadCertificate(final InputStream inputStream) throws CertificateException {
+        final CertificateFactory factory = CertificateFactory.getInstance("X.509");
+        return factory.generateCertificate(inputStream);
+    }
+
+    private KeyStore loadKeystore(final InputStream inputStream) throws CertificateException, IOException,
+            NoSuchAlgorithmException, KeyStoreException {
+        final KeyStore keyStore = KeyStore.getInstance("pkcs12");
+        keyStore.load(inputStream, PASSWORD);
+        return keyStore;
+    }
+
+    private PrivateKey loadPrivateKey(final InputStream inputStream) throws IOException, NoSuchAlgorithmException,
+            InvalidKeySpecException {
+        final byte[] rawKey = readAllBytes(inputStream);
+        final String key = new String(rawKey, US_ASCII);
+        /*
+         * Acknowledgement: Uses algorithm by Catalin Burcea. https://www.baeldung.com/java-read-pem-file-keys
+         * Bouncy Castle Library handles removing header and decoding, use instead?
+         */
+        final String privateKeyPEM = key
```

Comment:
looks like this code is [MIT licensed](https://github.com/eugenp/tutorials/blob/master/LICENSE) which we could use with license attribution (we use [notice file](https://github.com/apple/servicetalk/blob/main/servicetalk-http-api/NOTICE.txt) and [include the license in license directory](https://github.com/apple/servicetalk/tree/main/servicetalk-http-api/license) for that project). However since this code is simple and self contained can you plz use the following to avoid any complications due to attribution:

```diff
diff --git a/servicetalk-test-resources/src/test/java/io/servicetalk/test/resources/DefaultTestCertsTest.java b/servicetalk-test-resources/src/test/java/io/servicetalk/test/resources/DefaultTestCertsTest.java
index 985c32070..0874d63e8 100644
--- a/servicetalk-test-resources/src/test/java/io/servicetalk/test/resources/DefaultTestCertsTest.java
+++ b/servicetalk-test-resources/src/test/java/io/servicetalk/test/resources/DefaultTestCertsTest.java
@@ -17,9 +17,10 @@ package io.servicetalk.test.resources;
 
 import org.junit.jupiter.api.Test;
 
-import java.io.ByteArrayOutputStream;
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.security.KeyFactory;
 import java.security.KeyStore;
 import java.security.KeyStoreException;
@@ -32,8 +33,6 @@ import java.security.spec.InvalidKeySpecException;
 import java.security.spec.PKCS8EncodedKeySpec;
 import java.util.Base64;
 
-import static java.nio.charset.StandardCharsets.US_ASCII;
-
 class DefaultTestCertsTest {
     private static final char[] PASSWORD = "changeit".toCharArray();
 
@@ -96,30 +95,21 @@ class DefaultTestCertsTest {
 
     private PrivateKey loadPrivateKey(final InputStream inputStream) throws IOException, NoSuchAlgorithmException,
             InvalidKeySpecException {
-        final byte[] rawKey = readAllBytes(inputStream);
-        final String key = new String(rawKey, US_ASCII);
-        /*
-         * Acknowledgement: Uses algorithm by Catalin Burcea. https://www.baeldung.com/java-read-pem-file-keys
-         * Bouncy Castle Library handles removing header and decoding, use instead?
-         */
-        final String privateKeyPEM = key
-                .replace("-----BEGIN PRIVATE KEY-----", "")
-                .replaceAll(System.lineSeparator(), "")
-                .replace("-----END PRIVATE KEY-----", "");
-        byte[] encoded = Base64.getDecoder().decode(privateKeyPEM);
-        KeyFactory factory = KeyFactory.getInstance("RSA");
-        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(encoded);
-        return factory.generatePrivate(keySpec);
-    }
-
-    private byte[] readAllBytes(final InputStream inputStream) throws IOException {
-        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-        final byte[] data = new byte[1024];
-        int numBytesRead;
-        while ((numBytesRead = inputStream.read(data, 0, data.length)) != -1) {
-            outputStream.write(data, 0, numBytesRead);
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+            StringBuilder sb = new StringBuilder(2048);
+            String line = br.readLine();
+            if (!"-----BEGIN PRIVATE KEY-----".equals(line)) {
+                sb.append(line);
+            }
+            while ((line = br.readLine()) != null) {
+                sb.append(line);
+            }
+            int endKeyIndex = sb.lastIndexOf("-----END PRIVATE KEY-----");
+            String privateKeyPEM = endKeyIndex > 0 ? sb.substring(0, endKeyIndex) : sb.toString();
+            byte[] encoded = Base64.getDecoder().decode(privateKeyPEM);
+            KeyFactory factory = KeyFactory.getInstance("RSA");
+            PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(encoded);
+            return factory.generatePrivate(keySpec);
         }
-        outputStream.flush();
-        return outputStream.toByteArray();
     }
 }
```